Page guidelines

mcarrano opened the pull request:
This PR adds Page component guidelines that includes usage of page sections and the masthead. It also adds information about creating sticky headers and footers to pagination and toolbar.

Closes #2172 
Closes #2142 
Closes #714

patternfly-build commented:
PF3 preview: https://patternfly-org-pr-2297-v3.surge.sh/v3

mcarrano commented:
@redallen any idea why the v4 build is failing here?

redallen commented:
Looks like you didn't close an `<img>` tag:
> packages/v4/src/content/design-guidelines/components/page/page.md:59:1: Unexpected end of file in element, expected a corresponding MDX closing tag for `<img>`

Also there are merge conflicts.

redallen commented on packages/v4/src/content/design-guidelines/components/page/page.md:
```diff
@@ -2,4 +2,59 @@
+
+### Page with vertical navigation
+
+<img src="./img/page-vertical-nav.png" alt="page with vertical navigation" width="1596">
```
You have to close this `<img>` tag. Change to `<img ... />`

redallen commented on packages/v4/src/content/design-guidelines/components/page/page.md:
```diff
@@ -2,4 +2,59 @@
+
+### Page with horizontal navigation
+
+<img src="./img/page-horizontal-nav.png" alt="page with vertical navigation" width="1596">
```
Close this tag.

redallen commented on packages/v4/src/content/design-guidelines/components/page/page.md:
```diff
@@ -2,4 +2,59 @@
+## Masthead
+The masthead contains global properties such as logotype, navigation and settings in an organized fashion and it is accessible from all parts of the application. All pages should share a common masthead.
+
+<img src="./img/masthead.png" alt="page with vertical navigation" width="1623">
```
Close this tag.

mcarrano commented:
Thanks @redallen . For future reference, where did you find this message? I could not find anything that told me why the build failed even when I clicked into the Details.

redallen reviewed (changes requested):
Images can't be wider than 832px due to the max-width on the content you asked for. You should just set the widths to that.

redallen commented:
@mcarrano I just clicked the "details" button:
https://app.circleci.com/pipelines/github/patternfly/patternfly-org/2411/workflows/502a1cc6-9acd-4837-b9fd-68bbfcdefe3e/jobs/5869

These are the first lines on the failed step:
> yarn run v1.22.5
> $ yarn workspace patternfly-org-4 build && yarn copy:v4
> $ theme-patternfly-org build all
> write source files to src/generated
> /home/circleci/project/packages/v4/src/content/design-guidelines/components/page/page.md:61:1: Unexpected end of file in element, expected a corresponding MDX closing tag for `<img>` (41:1-41:80)

mcarrano commented:
@redallen I've fixed all of the image tags as you requested.

@mmenestr this is ready for final review. Besides adding documentation for the Page component, I added references to the use of sticky headers and footers in the Pagination and Toolbar design guidelines. For Pagination see callout for "Bottom pagination." For Toolbar see "Placement in a page."

mmenestr commented on packages/v4/src/content/design-guidelines/components/page/page.md:
```diff
@@ -2,4 +2,58 @@
+
+**2. Navigation:** This is your primary navigation. It should be consistent across pages. See the [Navigation design guidelines](/components/navigation/design-guidelines) for more information about navigation options and usage.
+
+**3. Breadcrumb (optional):** When breadcrumbs are included, they should be the topmost element in the header section. When breadcrumbs are included on a page, the top padding is reduced to 16px. See the [Breadcrumb design guidelines](/components/breadcrumb/design-guidelines) for more information about breadcrumb usage.
```
would add "the top padding is reduced to 16px, **from 24px**" since I don't think you mention top padding anywhere till this point!

mmenestr commented on packages/v4/src/content/design-guidelines/components/page/page.md:
```diff
@@ -2,4 +2,58 @@
+
+### Page with horizontal navigation
+
+<img src="./img/page-horizontal-nav.png" alt="page with horizontal navigation" width="832"/>
```
Wonder if there should still be the hamburger icon on the top left of this image, if there's no left nav?

mmenestr replied:
There's also a tiny grey line on the far right of the navigation that doesn't seem like it should be there haha

mmenestr commented on packages/v4/src/content/design-guidelines/components/page/page.md:
```diff
@@ -2,4 +2,58 @@
+## Masthead
+The masthead contains global properties such as logotype, navigation and settings in an organized fashion and it is accessible from all parts of the application. All pages should share a common masthead.
+
+<img src="./img/masthead.png" alt="masthead" width="832"/>
```
Small thing, but shouldn't the number in the notification badge be a part of the badge (ie: inside the blue box) and not outside of it?

mmenestr commented on packages/v4/src/content/design-guidelines/components/page/page.md:
```diff
@@ -2,4 +2,58 @@
+
+<img src="./img/masthead.png" alt="masthead" width="832"/>
+
+**1. Show/hide menu:**  provides a way for users to toggle vertical navigation. We recommend always having the menu icon visible when using vertical navigation. Show/hide menu is also used when application is being displayed in smaller viewport sizes, and it should stay visible in smaller viewport sizes.
```
I think there's a small word missing between "also used when ___ application is being displayed", guessing it should be "when **an** application"?

mmenestr replied:
@mcarrano just looked over the whole documentation and it looks ready to go apart from this small thing^

mcarrano replied:
@mmenestr I actually rewrote this callout entirely and hopefully made it more clear. Please review one more time and merge if you are good with the change.

mmenestr commented on packages/v4/src/content/design-guidelines/components/page/page.md:
```diff
@@ -2,4 +2,58 @@
+
+**2. Logotype:** The logotype area should clearly display the application name and the application logo if applicable. It is common to link the logotype to the application’s homepage. It is recommended to have the application name in .svg format for better performance.
+
+**3. Utility items:** Place any items in the utility area that you want to have accessible from all pages in the application. Utility items are right aligned after the user menu. The following are common items ordered from left to right. All items are optional depending on the needs of your application.
```
The sentence "Utility items are right aligned after the user menu" took me a second to understand. Would maybe change to something like "Utility items are right alighted on the masthead, and placed to the left of the user menu"

mcarrano replied:
Good point. Changed to "Utility items are right aligned and placed to the left of the user menu."

mmenestr commented on packages/v4/src/content/design-guidelines/components/page/page.md:
```diff
@@ -2,4 +2,58 @@
+
+* **Settings:**  The settings menu contains ways to configure and customize an application. Interacting with settings either takes users to a settings area or reveals a dropdown menu with more options. The settings menu is represented by the `fa-cog` icon.
+
+* **Help:** The help item reveals a dropdown menu. The menu items may include documentation links available to users, links to customer support resources, and/or instructions on how to get support. If the application has an [About modal](/components/about-modal), it should also be accessed here . The help menu is represented by the `pficon-pf-help` icon.
```
Extra space between "it should also be accessed here" and the period after it!

mmenestr replied:
I also think that "application launcher" and "about modal" should be sentence case when referred to in the sentences above

mcarrano replied:
I don't see this. Aren't they already sentence case?

mmenestr replied:
No, the first word is capitalized when it shouldn't be. So I mean change "About modal" to "about modal"

mcarrano replied:
Ah, OK. I misunderstood.

mmenestr reviewed (changes requested):
Looks great Matt! Just a few little tweaks and should be good to go!

mcarrano commented:
@mmenestr @redallen I've addressed all of your comments. Can you give this another review and merge if ready? Thanks.

mcarrano commented:
@redallen when you get a chance can you look at why this won't build? All I did was change some text in page.md and push to my origin repo. Thanks.

redallen commented:
@mcarrano It's failing because this branch doesn't yet have the optimizations I made in #2303 5 days ago. If you rebase or merge the latest master it should build fine.

mmenestr commented:
Looks good, merging! Thanks Matt!

patternfly-build commented:
Your changes have been released in: 
 - patternfly-org-4@4.2.92

Thanks for your contribution! :tada:
